main.tf

```
@@ -2,7 +2,7 @@ resource "aws_elasticache_replication_group" "redis" {
   engine = var.global_replication_group_id == null ? "redis" : null
 
   parameter_group_name = var.global_replication_group_id == null ? aws_elasticache_parameter_group.redis.name : null
-  subnet_group_name    = aws_elasticache_subnet_group.redis.name
+  subnet_group_name    = var.subnet_group_name == null ? try(var.subnet_group_name, aws_elasticache_subnet_group.redis[0].name) : var.subnet_group_name
```

the `try` function replaces the need for a conditional, null variables will be ignored
```suggestion
  subnet_group_name    = try(var.subnet_group_name, aws_elasticache_subnet_group.redis[0].name)
```

---

[7ff1e12](https://github.com/StuDocu/terraform-aws-elasticache-redis/pull/5/commits/7ff1e12b0421307931ea7b6b4166a7ebc4df4e97)